R/default-scoring-rules.R

```diff
@@ -0,0 +1,155 @@
+    return(possibilities[select])
+  } else {
+    assert_subset(select, allowed)
+    return(possibilities[select])
+  }
+}
+
+#' @title Scoring Rules for Binary Forecasts
+#' @description Helper function that returns a named list of default
+#' scoring rules suitable for binary forecasts.
+#'
+#' The default scoring rules are:
+#' - "brier_score" = [brier_score()]
+#' - "log_score" = [logs_binary()]
+#' @inherit select_rules params return
+#' @export
+#' @keywords info
+#' @examples
+#' rules_binary()
+#' rules_binary(select = "brier_score")
+#' rules_binary(exclude = "log_score")
+rules_binary <- function(select = "all", exclude = NULL) {
+  all <- list(
+    brier_score = brier_score,
+    log_score = logs_binary
+  )
+  selected <- select_rules(all, select, exclude)
+  return(selected)
+}
+
+
+#' @title Scoring Rules for Point Forecasts
+#' @description Helper function that returns a named list of default
+#' scoring rules suitable for point forecasts.
+#'
+#' The default scoring rules are:
+#' - "ae_point" = [ae()][Metrics::ae()]
+#' - "se_point" = [se()][Metrics::se()]
+#' - "ape" = [ape()][Metrics::ape()]
+#' @inherit select_rules params return
+#' @export
+#' @keywords info
+#' @examples
+#' rules_point()
+#' rules_point(select = "ape")
+rules_point <- function(select = "all", exclude = NULL) {
+  all <- list(
+    ae_point = Metrics::ae,
+    se_point = Metrics::se,
+    ape = Metrics::ape
+  )
+  selected <- select_rules(all, select, exclude)
+  return(selected)
+}
+
+
+#' @title Scoring Rules for Sample-Based Forecasts
+#' @description Helper function that returns a named list of default
+#' scoring rules suitable for forecasts in a sample-based format
+#'
+#' The default scoring rules are:
+#' - "mad" = [mad_sample()]
+#' - "bias" = [bias_sample()]
+#' - "dss" = [dss_sample()]
+#' - "crps" = [crps_sample()]
+#' - "log_score" = [logs_sample()]
+#' - "mad" = [mad_sample()]
+#' - "ae_median" = [ae_median_sample()]
+#' - "se_mean" = [se_mean_sample()]
+#' @inherit select_rules params return
+#' @export
+#' @keywords info
+#' @examples
+#' rules_sample()
+#' rules_sample(select = "mad")
+rules_sample <- function(select = "all", exclude = NULL) {
+  all <- list(
+    bias = bias_sample,
+    dss = dss_sample,
+    crps = crps_sample,
+    log_score = logs_sample,
+    mad = mad_sample,
+    ae_median = ae_median_sample,
+    se_mean = se_mean_sample
+  )
+  selected <- select_rules(all, select, exclude)
+  return(selected)
+}
+
+
+#' @title Scoring Rules for Quantile-Based Forecasts
+#' @description Helper function that returns a named list of default
+#' scoring rules suitable for forecasts in a quantile-based format
+#'
+#' The default scoring rules are:
+#' - "wis" = [wis]
+#' - "overprediction" = [overprediction()]
+#' - "underprediction" = [underprediction()]
+#' - "dispersion" = [dispersion()]
+#' - "bias" = [bias_quantile()]
+#' - "coverage_50" = [interval_coverage_quantile()]
+#' - "coverage_90" = \(...) \{
```

Comment:
internal uses function() - perhaps documenting this in a slightly more human readable way might be nice vs just reproducing the code?